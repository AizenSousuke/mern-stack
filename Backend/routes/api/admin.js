const express = require("express");
const router = express.Router();
const auth = require("../../middleware/auth");
const axios = require("axios");
const config = require("config");
const BusStop = require("../../models/BusStop");
const User = require("../../models/User");
const mongoose = require("mongoose");
const { check, validationResult } = require("express-validator");

router.get("/", (req, res) => {
	return res.status(200);
});

router.put("/UpdateBusStopList", auth, async (req, res) => {
	// Check if logged in user is an admin
	// Note that auth does not update when admin flag is set in db manually
	console.log("User is admin: " + req.user.IsAdmin);
	if (req.user.IsAdmin) {
		// Get all the bus stops from LTA
		let allBusStops = [];
		let arrayOfPromises = [];

		for (var pageSearched = 0; pageSearched < 11; pageSearched++) {
			arrayOfPromises.push(
				await axios
					.get(
						"http://datamall2.mytransport.sg/ltaodataservice/BusStops",
						{
							headers: {
								AccountKey: config.get("LTADataMallAPI"),
							},
							params: {
								$skip: pageSearched * 500,
							},
						}
					)
					.then((response) => {
						response.data.value.map((stops) => {
							allBusStops.push(stops);
						});
					})
					.catch((err) => {
						return res.status(500).json({ error: err.message });
					})
			);
		}

		Promise.all(arrayOfPromises)
			.then(async (data) => {
				// Update Mongoose DB with the data
				const session = await mongoose.startSession();
				try {
					session.startTransaction();

					// Drop the whole table first
					await BusStop.deleteMany({}).session(session);

					for (const stop of allBusStops) {
						await BusStop.findOneAndUpdate(
							{
								BusStopCode: stop.BusStopCode,
							},
							{
								RoadName: stop.RoadName,
								Description: stop.Description,
								Latitude: stop.Latitude,
								Longitude: stop.Longitude,
							},
							{ upsert: true }
						).session(session);
					}

					await session.commitTransaction();
					await session.endSession();

					return res.status(200).json({
						msg: "Successfully updated bus stop list",
						length: allBusStops.length,
					});
				} catch (err) {
					await session.abortTransaction();
					await session.endSession();
					return res
						.status(500)
						.json({ msg: "Server error", error: err.message });
				}
			})
			.catch((err) => {
				return res
					.status(500)
					.json({ msg: "Server error", error: err.message });
			});
	} else {
		return res.status(403).json({ msg: "You must be an admin to do this" });
	}
});

router.patch(
	"/setAdmin",
	[
		check(
			"Email",
			"Please provide the email address of the user that you want to set an admin as"
		).isEmail(),
		check("IsAdmin", "Please set true or false").isBoolean(),
	],
	auth,
	async (req, res) => {
		try {
			const user = req.user;
			if (!user.IsAdmin) {
				return res
					.status(403)
					.json({ msg: "You must be an admin to do this" });
			}
			const errors = validationResult(req);
			if (!errors.isEmpty()) {
				return res.status(422).json({ msg: errors.array() });
			}
			const { Email, IsAdmin } = req.body;
			const updatedUser = await User.findOneAndUpdate(
				{ Email: Email },
				{ IsAdmin: IsAdmin }
			);
			if (!updatedUser) {
				return res.status(404).json({ msg: "No user found" });
			}

			return res.status(200).json({ msg: "Successfully set admin" });
		} catch (err) {}
	}
);

module.exports = router;
