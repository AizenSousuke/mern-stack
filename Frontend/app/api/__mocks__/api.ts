export const GetBusStopByCode = jest.fn().mockResolvedValue({
    busStop: {
        Description: "Sample Description",
        RoadName: "Sample Road Name",
        BusStopCode: "44221",
        Location: [
            103.771459794786,
            1.37656055754166
        ],
    }
});
export const GetBusStop = jest.fn().mockResolvedValue({
    data: {
        BusStopCode: "44221",
        Services: [
            {
                ServiceNo: "171",
                Operator: "TTS",
                NextBus: {
                    OriginCode: "59009",
                    DestinationCode: "59009",
                    EstimatedArrival: "2023-02-15T09:12:27+08:00",
                    Latitude: "1.3766951666666667",
                    Longitude: "103.77152733333334",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus2: {
                    OriginCode: "59009",
                    DestinationCode: "59009",
                    EstimatedArrival: "2023-02-15T09:21:10+08:00",
                    Latitude: "1.402927",
                    Longitude: "103.7734205",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus3: {
                    OriginCode: "59009",
                    DestinationCode: "59009",
                    EstimatedArrival: "2023-02-15T09:30:16+08:00",
                    Latitude: "1.412433",
                    Longitude: "103.78906516666666",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                }
            },
            {
                ServiceNo: "184",
                Operator: "SMRT",
                NextBus: {
                    OriginCode: "44989",
                    DestinationCode: "44989",
                    EstimatedArrival: "2023-02-15T09:20:42+08:00",
                    Latitude: "1.3856701666666666",
                    Longitude: "103.757206",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus2: {
                    OriginCode: "44989",
                    DestinationCode: "44989",
                    EstimatedArrival: "2023-02-15T09:25:37+08:00",
                    Latitude: "1.390941",
                    Longitude: "103.75635666666666",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "BD"
                },
                NextBus3: {
                    OriginCode: "44989",
                    DestinationCode: "44989",
                    EstimatedArrival: "2023-02-15T09:36:41+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                }
            },
            {
                ServiceNo: "960",
                Operator: "SMRT",
                NextBus: {
                    OriginCode: "02099",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:21:53+08:00",
                    Latitude: "1.3340475",
                    Longitude: "103.81220866666666",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus2: {
                    OriginCode: "02099",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:24:38+08:00",
                    Latitude: "1.3239830000000001",
                    Longitude: "103.82665183333333",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus3: {
                    OriginCode: "02099",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:38:26+08:00",
                    Latitude: "1.3045368333333334",
                    Longitude: "103.85095",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                }
            },
            {
                ServiceNo: "963",
                Operator: "TTS",
                NextBus: {
                    OriginCode: "46009",
                    DestinationCode: "14009",
                    EstimatedArrival: "2023-02-15T09:14:57+08:00",
                    Latitude: "1.3743405",
                    Longitude: "103.77832283333333",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus2: {
                    OriginCode: "46009",
                    DestinationCode: "14009",
                    EstimatedArrival: "2023-02-15T09:28:01+08:00",
                    Latitude: "1.4321488333333334",
                    Longitude: "103.77419216666667",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus3: {
                    OriginCode: "46009",
                    DestinationCode: "14009",
                    EstimatedArrival: "2023-02-15T09:29:58+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                }
            },
            {
                ServiceNo: "966",
                Operator: "TTS",
                NextBus: {
                    OriginCode: "46009",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:15:05+08:00",
                    Latitude: "1.3777718333333333",
                    Longitude: "103.7766495",
                    VisitNumber: "1",
                    Load: "SDA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus2: {
                    OriginCode: "46009",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:25:25+08:00",
                    Latitude: "1.4253149999999999",
                    Longitude: "103.77101633333334",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus3: {
                    OriginCode: "46009",
                    DestinationCode: "46009",
                    EstimatedArrival: "2023-02-15T09:30:01+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                }
            },
            {
                ServiceNo: "972M",
                Operator: "SMRT",
                NextBus: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:29:05+08:00",
                    Latitude: "1.3868446666666667",
                    Longitude: "103.7649695",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus2: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:38:47+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                },
                NextBus3: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:56:47+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "DD"
                }
            },
            {
                ServiceNo: "973",
                Operator: "SMRT",
                NextBus: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:14:16+08:00",
                    Latitude: "1.3789708333333333",
                    Longitude: "103.7694475",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus2: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:25:57+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus3: {
                    OriginCode: "45009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:39:57+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                }
            },
            {
                ServiceNo: "976",
                Operator: "SMRT",
                NextBus: {
                    OriginCode: "44009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:21:31+08:00",
                    Latitude: "1.3882765",
                    Longitude: "103.7642615",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus2: {
                    OriginCode: "44009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:40:18+08:00",
                    Latitude: "1.3828756666666666",
                    Longitude: "103.74281433333333",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                },
                NextBus3: {
                    OriginCode: "44009",
                    DestinationCode: "45009",
                    EstimatedArrival: "2023-02-15T09:53:41+08:00",
                    Latitude: "0",
                    Longitude: "0",
                    VisitNumber: "1",
                    Load: "SEA",
                    Feature: "WAB",
                    Type: "SD"
                }
            }
        ]
    }
});


export default {};