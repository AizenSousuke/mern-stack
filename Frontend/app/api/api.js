import * as secrets from "../../secrets.json";
import axios from "axios";
import * as SQLite from "expo-sqlite";

/**
 * Strings for use in SQL
 */
const dataKey = {
	BusStopList: "BusStopList",
	Settings: "Settings",
	GoingOut: "GoingOut",
	GoingBack: "GoingBack",
};
/**
 * Store the bus stops here
 */
const databaseName = "sgbus.db";
/**
 * Store user settings here
 *
 * Should store the Going Out List and the Going Home List in 2 entries
 * For future use, the going out list can be a list of list (with the bus services saved)
 * I.e,
 * Going Out
 * [{
 * 		"busStop": ["service_966", "service_911"]
 * }]
 */
const settingsDatabaseName = "settings.db";
/**
 * Api key to be provided from a secrets.json file
 */
const apiKey = secrets.apiKey;
/**
 * Headers to be set for the requests
 */
const header = {
	AccountKey: apiKey,
	Accept: "application/json",
};
var BusArrivalURL =
	"http://datamall2.mytransport.sg/ltaodataservice/BusArrivalv2";
var BusStopsURL = "http://datamall2.mytransport.sg/ltaodataservice/BusStops";

var db = SQLite.openDatabase(databaseName);

export const getBusArrival = async (code, serviceNumber = null) => {
	var data = await axios
		.get(BusArrivalURL, {
			headers: header,
			params: {
				BusStopCode: code,
				ServiceNo: serviceNumber,
			},
		})
		.then((res) => {
			// console.log(res.data);
			return res.data;
		});

	return data;
};

export const getBusStops = async (skip = null) => {
	var data = await axios
		.get(BusStopsURL, {
			headers: header,
			params: {
				$skip: skip * 500,
			},
		})
		.then((res) => {
			// console.log(res.data);
			return res.data;
		});

	return data;
};

export const getAllBusStops = async () => {
	// var data = await Promise.all[]
};

/**
 * Create Bus Stop Table if it doesn't exists
 */
export const BusStopTableCheck = () => {
	return new Promise((resolve, reject) => {
		db.transaction((tx) => {
			tx.executeSql(
				`
					CREATE TABLE IF NOT EXISTS ${dataKey.BusStopList} (
						Id INTEGER PRIMARY KEY AUTOINCREMENT,
						Data nvarchar NULL,
						LastUpdated datetime2 DEFAULT CURRENT_TIMESTAMP
					);
				`,
				[],
				(res, result) => {
					tx.executeSql(
						`
						SELECT * FROM ${dataKey.BusStopList};
					`,
						[],
						(res, result) => {
							resolve(
								JSON.stringify(result.rows._array, null, "\t")
							);
						},
						(err) => {
							console.log("Error: %s", err);
							reject(null);
						}
					);
				},
				(err) => {
					console.log("Error: %s", err);
					reject(null);
				}
			);
		});
	});
};

export const settingsTableCheck = () => {
	return new Promise((resolve, reject) => {
		db.transaction((tx) => {
			tx.executeSql(
				`
				CREATE TABLE IF NOT EXISTS ${dataKey.Settings} (
					Id INTEGER PRIMARY KEY,
					Data nvarchar NULL,
					LastUpdated datetime2 DEFAULT CURRENT_TIMESTAMP
				);
			`,
				[],
				(res, result) => {},
				(err) => {
					console.error("Error: %s", err);
					reject(null);
				}
			);
		});
	});
};

export const DeleteTable = (value) => {
	try {
		return new Promise((resolve, reject) => {
			db.transaction((tx) => {
				tx.executeSql(
					`
					DROP TABLE IF EXISTS ${value ?? dataKey.BusStopList}
					`,
					[],
					(res, result) => {
						console.log("Deleted table: %s", value);
						resolve(true);
					},
					(err, result) => {
						console.error(result.message);
						reject(false);
					}
				);
			});
		});
	} catch (error) {
		console.log(error);
	}
};

/**
 * Stores a data into Bus Stop List Table
 * @param {any} value Data to store
 * @returns true if data is successfully saved
 */
export const storeData = (value) => {
	try {
		return new Promise((resolve, reject) => {
			db.transaction(
				(tx) => {
					// tx.executeSql(`
					// DROP TABLE BusStopList
					// `);

					BusStopTableCheck(tx);

					tx.executeSql(
						`
						INSERT INTO ${dataKey.BusStopList} (
							Data
						) VALUES (
							?
						)
					`,
						[JSON.stringify(value)],
						(tx, res) => {
							// console.log("Value: " + JSON.stringify(value));
							// console.log("Res: " + JSON.stringify(res));
							resolve(true);
						}
					);
				},
				(err) => {
					console.error("Error in storing to SQLite " + err.message);
					reject(false);
				},
				() => {
					console.log("Successfully stored data in SQLite");
				}
			);
		});
	} catch (error) {
		console.log("Saving Error: " + error);
	}
};

/**
 * Get data from the Database
 * @param {string} value If value is provided, it will search the SQL table for that value
 */
export const getData = (value = null) => {
	try {
		return new Promise((resolve, reject) => {
			db.transaction(
				(tx) => {
					if (value == null) {
						console.log("Value provided is null");
						tx.executeSql(
							`
							SELECT Data FROM ${dataKey.BusStopList};
						`,
							[],
							(tx, res) => {
								resolve(res.rows._array);
							},
							(err) => {
								console.log("Error getting data ", err);
								reject(null);
							}
						);
					} else {
						tx.executeSql(
							`
							SELECT Data FROM ${dataKey.BusStopList}
							WHERE 
								Data LIKE '%' + ? + '%';
						`,
							[value],
							(tx, res) => {
								// res.rows._array.forEach((item) => {
								// 	console.log("Items: %s", item);
								// });
								// console.log("Value: " + JSON.stringify(value));
								// console.log("Res: " + JSON.stringify(res));
								resolve(res.rows._array);
							},
							(err) => {
								console.log(
									"Error getting filtered data ",
									err
								);
								reject(null);
							}
						);
					}
				},
				(err) => {
					console.error(
						"Error in getting data from SQLite " + err.message
					);
					reject(null);
				},
				() => {
					console.log("Successfully read data from SQLite");
				}
			);
		});
	} catch (error) {
		console.log("Get Data Error: " + error);
	}
};

/**
 * Gets the last updated date of the bus stops in order to update
 * it if required
 */
export const getLastUpdatedDate = () => {
	try {
		return new Promise((resolve, reject) => {
			db.transaction(
				(tx) =>
					tx.executeSql(
						`
							SELECT LastUpdated 
							FROM ${dataKey.BusStopList}
							ORDER BY LastUpdated DESC
						`,
						[],
						(tx, res) => {
							if (res.rows._array.length > 0) {
								// console.log(
								// 	"Last updated date: ",
								// 	res.rows.item(0).LastUpdated
								// );
								resolve(res.rows.item(0).LastUpdated);
							} else {
								resolve(null);
							}
						},
						(err) => {
							console.log("Error executing sql {0}", err);
							reject(null);
						}
					),
				(err) => {
					console.log("Error reading transaction {0}", err);
					reject(null);
				},
				() => {
					console.log("Successfully read transaction");
					resolve(null);
				},
				(err) => {
					console.log("Error %s", err);
					reject(null);
				}
			);
		});
	} catch (error) {
		console.log("Error getting last updated date from db {0}", error);
	}
};

export const DeleteBusStopList = () => {
	try {
		return new Promise((resolve, reject) => {
			db.transaction((tx) =>
				tx.executeSql(
					`
					DELETE FROM ${dataKey.BusStopList} WHERE Data NOT NULL
					`,
					[],
					(res) => {
						console.log("Deleted items from table");
						resolve(true);
					},
					(err) => {
						console.log("Error deleting items from table ", err);
						reject(false);
					}
				)
			);
		});
	} catch (error) {}
};

// Sample function
export const SampleFunctionThatReturnsSomething = (param) => {
	return new Promise((resolve, reject) => {
		db.transaction((tx) => {
			tx.executeSql(
				`SQL STATEMENT`,
				[parameters],
				(transaction, resultSet) => {
					resolve(resultSet);
				},
				(transaction, error) => {
					reject(error);
				}
			);
		});
	});
};
