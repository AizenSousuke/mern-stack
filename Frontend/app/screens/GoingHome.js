import React, { useState, useEffect } from "react";
import { View, Text } from "react-native";

export default function GoingHome() {
	return (
		<View>
			
		</View>
	);
}
