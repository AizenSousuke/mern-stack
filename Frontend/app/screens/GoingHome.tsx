import React, { useState } from "react";
import { RefreshControl, Text } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import BusStopSaved from "../components/BusStopSaved";

const GoingHome = (props: any) => {
	const [refreshing, setRefreshing] = useState(false);

	return (
		<ScrollView
			refreshControl={
				<RefreshControl
					refreshing={refreshing}
					onRefresh={() => {
						console.log("Refreshing");
						props.updateSettings();
					}}
				></RefreshControl>
			}
		>
			{props?.settings?.GoingHome?.map((busStop: any, index: number) => (
				<BusStopSaved key={index} code={busStop} GoingOut={false} />
			))}
		</ScrollView>
	);
};

export default GoingHome;
