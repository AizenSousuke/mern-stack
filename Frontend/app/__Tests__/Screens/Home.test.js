import ReactTestRenderer, { create } from "react-test-renderer";
import { render } from "@testing-library/react-native";
import React from "react";
import GoingHome from "../../screens/GoingHome";
import { GetBusStopByCode, GetBusStop } from "../../api/api";

jest.mock("react-native-gesture-handler", () => {
	// eslint-disable-next-line global-require
	const View = require("react-native/Libraries/Components/View/View");
	return {
		Swipeable: View,
		DrawerLayout: View,
		State: {},
		// This is needed to not error out below in the tests
		ScrollView: View,
		Slider: View,
		Switch: View,
		TextInput: View,
		ToolbarAndroid: View,
		ViewPagerAndroid: View,
		DrawerLayoutAndroid: View,
		WebView: View,
		NativeViewGestureHandler: View,
		TapGestureHandler: View,
		FlingGestureHandler: View,
		ForceTouchGestureHandler: View,
		LongPressGestureHandler: View,
		PanGestureHandler: View,
		PinchGestureHandler: View,
		RotationGestureHandler: View,
		/* Buttons */
		RawButton: View,
		BaseButton: View,
		RectButton: View,
		BorderlessButton: View,
		/* Other */
		FlatList: View,
		gestureHandlerRootHOC: jest.fn(),
		Directions: {},
		Icon: View
	};
});

jest.mock("../../api/api.js", () => ({
	GetBusStopByCode: jest.fn(),
	GetBusStop: jest.fn()
}));

GetBusStopByCode.mockImplementation((code) => {
	Promise.resolve({ Description: "", RoadName: "", Code: code });
});

GetBusStop.mockImplementation((code) => {
	Promise.resolve(null);
})

describe("Home", () => {
	it("renders properly", () => {
		// const testRenderer = ReactTestRenderer.create(<GoingHome />);
		// expect(testRenderer.toJSON()).toMatchSnapshot();

		// expect(render(<GoingHome />).toJSON()).toMatchSnapshot();

		let root = create(<GoingHome />);
		expect(root.toJSON()).toMatchSnapshot();
	});

	it("renders list of bus stops properly", () => {
		const settings = {
			GoingHome: [44229],
		};
		let root = create(<GoingHome settings={settings} />);
		expect(root.toJSON()).toMatchSnapshot();
	});
});
