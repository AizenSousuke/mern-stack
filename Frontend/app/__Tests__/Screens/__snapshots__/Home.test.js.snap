// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders list of bus stops properly 1`] = `
<View
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomWidth": 0.5,
            "borderColor": "#bcbbc1",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="padView"
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            Bus Stop Name
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 15,
              }
            }
            testID="listItemTitle"
          >
            Address
             (
            44229
            )
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 16,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View>
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 0, 0, .4)",
                    "height": "100%",
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Overlay__backdrop"
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 3,
                      "padding": 10,
                      "shadowColor": "rgba(0, 0, 0, .3)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowRadius": 4,
                    }
                  }
                >
                  <View>
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "borderColor": "#bcbbc1",
                            "flexDirection": "row",
                            "padding": 14,
                          }
                        }
                        testID="padView"
                      >
                        <Text
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#242424",
                              "fontSize": 17,
                            }
                          }
                          testID="listItemTitle"
                        >
                          What do you want to do?
                        </Text>
                      </View>
                    </View>
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "borderColor": "#bcbbc1",
                            "flexDirection": "row",
                            "padding": 14,
                          }
                        }
                        testID="padView"
                      >
                        <Text
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#242424",
                              "fontSize": 15,
                            }
                          }
                          testID="listItemTitle"
                        >
                          Delete
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        forwardedRef={[Function]}
        onLayout={[Function]}
        style={Object {}}
      >
        <Text>
          No Data
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Home renders properly 1`] = `
<View
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
/>
`;
